### PR TITLE
few fixes

### PR DESCRIPTION
**Pull request by Skarm**
replaced defer
sorting structure fields to reduce memory

### CODE REVIEW

**Comment by FZambia** [comment]
@Skarm thanks, this is a bunch of useful things, in general I agree with everything except refactoring `defers` - since [defers are inlined in Go 1.14](https://rakyll.org/inlined-defers/) - don't think we should sacrifice readability.

File: engine_memory_test.go [hunk 1]
```diff
@@ -400,9 +400,9 @@ var recoverTests = []struct {
 	HistorySize     int
 	HistoryLifetime int
 	NumPublications int
-	SinceSeq        uint32
 	NumRecovered    int
 	Sleep           int
+	SinceSeq        uint32
```
Comment (FZambia) [inline engine_memory_test.go:405]: This change have broken tests since order is important below.

Comment (Skarm) [reply engine_memory_test.go:405]: oops, I'll keep it on mind ;)

**Comment by Skarm** [comment]
make new PR?

**Comment by FZambia** [comment]
You can update this one removing defer changes and fixing order in tests or create new one - just do what is more comfortable for you.

**Pull request closed without merging**